GVFS/GVFS.FunctionalTests.LockHolder/AcquireGVFSLock.cs

```diff
@@ -45,6 +41,45 @@ public void Execute()
             }
         }
 
+        private static bool TryGetGVFSEnlistmentRootImplementation(string directory, out string enlistmentRoot, out string errorMessage)
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                return MacPlatform.TryGetGVFSEnlistmentRootImplementation(directory, out enlistmentRoot, out errorMessage);
+            }
+            else
+            {
+                enlistmentRoot = null;
+
+                string finalDirectory;
+                if (!WindowsFileSystem.TryGetNormalizedPathImplementation(directory, out finalDirectory, out errorMessage))
+                {
+                    return false;
+                }
+
+                enlistmentRoot = Paths.GetRoot(finalDirectory, GVFSConstants.DotGVFS.Root);
+                if (enlistmentRoot == null)
+                {
+                    errorMessage = $"Failed to find the root directory for {GVFSConstants.DotGVFS.Root} in {finalDirectory}";
+                    return false;
+                }
+
+                return true;
+            }
+        }
+
+        private static string GetNamedPipeNameImplementation(string enlistmentRoot)
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                return MacPlatform.GetNamedPipeNameImplementation(enlistmentRoot);
+            }
+            else
+            {
+                return "GVFS_" + enlistmentRoot.ToUpper().Replace(':', '_');

```

Comment:
Not able to use WindowsPlatform here - because of its dependency on WindowsIdentity (and also kernel32.dll). Let me know if there is an alternate approach.